firmware/Core/Src/Controller.c

```diff
@@ -103,9 +103,43 @@ static Controller_State_t LogData_State(void)
     return CTRL_DO_MATH;
 }
 
+float savedTemps[];
+float loopCounter;
+
+
 static Controller_State_t DoMath_State(void)
 {
-    return CTRL_ACTUATE_FRIDGE;
+    float temps[5] = {t1, t2, t3, t4, t5}; //5 temp probe readings
+    float temp_diffs[5]; //to store deltas between probe readings and avg temp
+    float avgTemp = 0; //mean temperature
+    float fan_treshold = 1; //desired fan treshold
+    int actuate[2] = {0, 0}; //first position is fan, second is compressor
```

Comment:
Take a look at Actuators.h You'll find the definition to a struct. At the top of this (Controller.c) file, under the variables section, you'l find "static ActuatorCommands_t ActuatorCommands;". That's the instance of the struct you should populate. Once you do, you're done. The next state knows to read this variable to get the data you populated it with.


(Sidenote: I'll be moving the definition of that struct to Common.h, makes way more sense to have it there, but that doesen't affect you stuff in any way. Controller knows about that struct either way).